M17_spec.tex

```diff
@@ -1436,150 +1494,33 @@ \subsection{Encryption Types}
 
-The random part of the nonce value should be generated with a hardware random number generator or any other cryptographically secure method of generating random values.
-To prevent replay attacks, a 32-bit timestamp is included in the cryptographic nonce field. The structure of the 128-bit counter is shown in Table~\ref{tab:aes_counter}. The timestamp represents the number of seconds since 2020-01-01T00:00:00Z (midnight UTC, January 1, 2020).
+Up to four Text Data blocks compose a complete message with a maximum length of 52 bytes. Each block may contain up to 13 bytes of UTF-8 encoded text, and is padded with null characters to fill any unused space at the end of the last used Text Data block.
 
-\paragraph{128 bit counter structure}
+The Control Byte is split into two 4-bit fields. The most significant four bits are a bit map of the message length indicating how many Text Data blocks are required for a complete message. There is one bit per used Text Data block, with $0001_2$ used for one block, $0011_2$ for the two, $0111_2$ for three, and $1111_2$ for four.
 
-FN field sets the most significant 16 bits of the counter, with the 32-bit least significant part holding the timestamp. The remaining 80-bit portion is filled with random data, re-generated per transmission.
+The least significant four bits indicate which of the Text Data blocks this text corresponds to. It is $0001_2$ for the first, $0010_2$ for the second, $0100_2$ for the third, and $1000_2$ for the fourth. Any received Control Byte is OR-ed together by the receiving station, and once the most significant and least significant four bits are the same, a complete message has been received.
 
-\begin{table}[h]
-	\centering
-	\begin{tblr}{
-		colspec={lcl},
-		}
-		\hline
-		Field & Length & Description \\
-		\hline
-		Timestamp & 32 bits & Seconds since 2020-01-01T00:00:00Z \\
-		Random Data & 80 bits & Random data \\
-		Frame Number & 16 bits & Incrementing Frame Number \\
-		\hline[2px]
-	\end{tblr}
-	\caption{AES Counter Contents}
-	\label{tab:aes_counter}
-\end{table}
+It is up to the receiver to decide how to display this message. It may choose to wait for all of the Text Data to be received, or display the parts as they are received. It is not expected that the data in the text field changes during the course of a transmission.
 
-\begin{quote}
-	\textbf{WARNING} In CTR mode, AES encryption is malleable. That is, an attacker can change the contents of the encrypted message without decrypting it. This means that recipients of AES-encrypted data must not trust that the data is authentic. Users who require that received messages are proven to be exactly as-sent by the sender should use an appropriate digital signature algorithm, as described below.
-\end{quote}
+If there is no data to be transferred, the Control Byte should be set to zero.
 
```

Comment:
TShould there should be some reference to the AES Encryption IV here, pointing to where the actual data format is defined?